Bug 1145280 - Create team landing pages

bobsilverberg opened the pull request:
This implements the team landing pages as discussed in bug 1145280.

@glogiotatidis | @akatsoulas r? please and thank you

glogiotatidis commented on oneanddone/tasks/filters.py:
```diff
@@ -36,6 +36,27 @@ class Meta:
+            ('', 'Any'),
+            (Task.BEGINNER, 'Beginner'),
+            (Task.INTERMEDIATE, 'Intermediate'),
+            (Task.ADVANCED, 'Advanced')),
```
Localization?

glogiotatidis commented on oneanddone/tasks/filters.py:
```diff
@@ -36,6 +36,27 @@ class Meta:
+        required=False)
+
+    execution_time = django_filters.ChoiceFilter(
+        choices=(('', 'Any'), (15, 15), (30, 30), (45, 45), (60, 60)),
```
Localizatoin?

glogiotatidis commented on oneanddone/tasks/filters.py:
```diff
@@ -43,6 +64,14 @@ class TasksFilterSet(django_filters.FilterSet):
+        choices=(
+            (Task.BEGINNER, 'Beginner'),
+            (Task.INTERMEDIATE, 'Intermediate'),
+            (Task.ADVANCED, 'Advanced')),
```
Localization?

glogiotatidis commented on oneanddone/tasks/forms.py:
```diff
@@ -198,3 +199,31 @@ class TaskInvalidationCriterionForm(forms.Form):
+
+class TeamForm(forms.ModelForm):
+    url_code = forms.RegexField(
+        label=_("Team url suffix:"),
```
You need `_lazy` for these

glogiotatidis commented on oneanddone/tasks/models.py:
```diff
@@ -312,11 +311,40 @@ def __unicode__(self):
+    @classmethod
+    def get_team_by_id_or_url_code(self, kwargs):
+        url_code = kwargs.get('url_code')
+        if url_code is None:
```
I'd use the more generic and simple `if not url_code`

glogiotatidis commented on oneanddone/tasks/models.py:
```diff
@@ -312,11 +311,40 @@ def __unicode__(self):
+        if url_code is None:
+            try:
+                return TaskTeam.objects.get(id=kwargs.get('pk'))
+            except ObjectDoesNotExist:
```
I've been bitten in the past while using the generic ObjectDoesNotExist exception and catching one that was raised for another object than the one I intended it to. This of course is a very simple try/except nevertheless I'd still go with TaskTeam.DoesNotExist

bobsilverberg replied:
Thanks @glogiotatidis. I didn't even know about `TaskTeam.DoesNotExist` as a more specific version of `ObjectDoesNotExist`. Cool!

glogiotatidis commented:
Just a few nits. Nice work! r+wc

bobsilverberg commented:
Thanks for the review @glogiotatidis! I'm still not used to thinking about l10n all the time, so I appreciate you finding those issues. I addressed all of your comments and will merge the PR.
